cmd/wit-bindgen-go/cmd/generate/generate.go

```diff
@@ -140,25 +148,44 @@ func parseFlags(cmd *cli.Command) (*config, error) {
 		cmd.Bool("versioned"),
 		cmd.Bool("force-wit"),
 		path,
+		verbose,
 	}, nil
 }
 
+// Generate and write the specified Go packages w/ provided bindgen config
+//
+// Parameters:
+//   - packages: A slice of pointers to `gen.Package`s to generate
+//   - cfg: A pointer to a `config` struct
+//
+// Returns:
+//   - error: An error if the package writing process fails, or nil if all
+//     packages are written successfully.
 func writeGoPackages(packages []*gen.Package, cfg *config) error {
-	fmt.Fprintf(os.Stderr, "Generated %d package(s)\n", len(packages))
+	var stderrWriter io.Writer
```

Comment:
Consider using package log to configure a few different loggers: debug, info (-v), and error.

Reply:
Sure will update this to use it!

Reply:
OK so what I did here was to use verbose to switch between `os.Stderr` and `io.Discard`, but were you expecting a `--log-level` type argument there and a more complicated setup? 

From the earlier discussion in the issue, I though that a simple setup like not printing anything if verbose wasn't set was enough). If we *do* want to do a bit more complicated logging, as we can't really set the level with the built in `log`, is there a pattern that is normally used? I could of course write something ad-hoc, but would prefer not to!

Reply:
I've updated the code -- reproducing here for easier review:

```
func main() {
	cmd := makeCmd()

	// If verbose was specified, use set global logger appropriately
	if cmd.Bool("verbose") {
		log.SetOutput(os.Stderr)
	} else {
		log.SetOutput(io.Discard)
	}

	err := cmd.Run(context.Background(), os.Args)
	if err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}
```

With that change I removed the manual `io` stuff from `generate.go`, and now we're using the built-in `log`

Reply:
**suggest**: we can set debug, info and error loggers as the following:

```Go

var (
	debugLogger *log.Logger
	infoLogger  *log.Logger
	errorLogger *log.Logger
)

func initLoggers(verbose bool) {
	errorLogger = log.New(os.Stderr, "ERROR: ", log.Ldate|log.Ltime|log.Lshortfile)
        infoLogger = log.New(os.Stderr, "INFO: ", log.Ldate|log.Ltime)
	if verbose {	
		debugLogger = log.New(os.Stderr, "DEBUG: ", log.Ldate|log.Ltime|log.Lshortfile)
	} else {
		debugLogger = log.New(io.Discard, "", 0) // Discard debug logs
	}
}
```

Reply:
Ah this is a wider change than I was confident to make... I'd certainly like to just use it if @ydnar is onboard since this is the ~last outstanding issue!